site/shared/src/main/scala/shared/User.scala

```diff
@@ -23,3 +23,26 @@ case class UserProgress(
   args:         Option[String],
   succeeded:    Boolean
 )
+
+case class OverallUserProgressItem(
+  libraryName: String,
+  sections:    Int,
+  completed:   Int
+)
+
+case class OverallUserProgress(libraries: List[OverallUserProgressItem])
+
+case class LibrarySectionArgs(
```

Comment:
In this case we need a list with the state of every exercise progress for the given library and section. For each exercise we'll need its method name, arguments, and version.